_stbt/core.py

```diff
@@ -2527,21 +2526,33 @@ def _tesseract(frame, region, mode, lang, _config,
     else:
         region = intersection
 
-    return (_tesseract_subprocess(_crop(frame, region), mode, lang,
-                                  _config, user_patterns, user_words),
+    return (_tesseract_subprocess(_crop(frame, region), mode, lang, _config,
+                                  user_patterns, user_words, text_color),
             region)
 
 
 @imgproc_cache.memoize({"tesseract_version": str(_tesseract_version()),
-                        "version": "25"})
+                        "version": "27"})
 def _tesseract_subprocess(
-        frame, mode, lang, _config, user_patterns, user_words):
+        frame, mode, lang, _config, user_patterns, user_words, text_color):
+
     # We scale image up 3x before feeding it to tesseract as this
     # significantly reduces the error rate by more than 6x in tests.  This
     # uses bilinear interpolation which produces the best results.  See
     # http://stb-tester.com/blog/2014/04/14/improving-ocr-accuracy.html
     outsize = (frame.shape[1] * 3, frame.shape[0] * 3)
-    subframe = cv2.resize(frame, outsize, interpolation=cv2.INTER_LINEAR)
+    frame = cv2.resize(frame, outsize, interpolation=cv2.INTER_LINEAR)
+
+    if text_color is not None:
+        # Calculate distance of each pixel from `text_color`, then discard
+        # everything further than `threshold` distance away.
+        diff = frame - text_color
```

Comment:
> >         diff = frame - text_color
> What about overflow? Presumably these are both uint8?

`frame` is `uint8`. `text_color` is just a (non-numpy) tuple.

If they were both `uint8` numpy arrays the result would be `uint8` and there would be overflow. Since `text_color` isn't a numpy array the result is `int64`.

I'll push a fix to make this more explicit by using `diff = numpy.subtract(frame, text_color, dtype=numpy.int32)`.